### DISCUSSION TITLE
Selecting lines consistency

### DISCUSSION THREAD

**Discussion by CptPotato** [helix-editor/helix#835] [Q&A] [answered]
One minor annoyance that I keep running into is the inconsistency of selecting empty lines with `x`.
Take this for example:

```rust
fn foo() {
    abc();

    xyz();
}
```

I want to delete the (empty) line between `abc()` and `xyz()`.
I can use `xd` to delete lines as usual. However, this only works if the line happens to have some indentation. Otherwise it will instead add the following line to the selection and delete more than I wanted.

I know that this is because a cursor is essentially a one character wide selection, so an empty line is already selected by simply moving the cursor there. But I'm often thrown off by this and missing an option to "just delete a line" without having to stop and look at what's currently selected.

Is it feasible to prevent `x` from selecting the next line on the first time it's executed? Or are there any alternative commands to do this?

**Comment by CptPotato** [comment, marked as answer]
Looking through the documentation, I saw `X` (uppercase) which only extends to the line bounds so it works for the case I mentioned above. `Xd` is a bit awkward to enter quickly, though, so I switched `X` and `x` which is a decent workaround:
Selecting a single line is fast and consistent and having to hold the <kbd>Shift</kbd> key for extending the selection is good enough (I usually extend multiple times in a row so I can just keep <kbd>Shift</kbd> pressed)

I'm still open to other thoughts and options, though.

Lastly, another inconsistency I noticed is that selecting two or more lines with `<n>x` always works as expected - i.e. `3x` will always select three lines, even if the cursor starts on an empty line. (I actually like this behavior, even though it doesn't currently reflect what happens when you press `x` three times)

**Reply by pickfire** [reply]
I had the same issue with this as well. For quite a few times I don't know whether the current line have whitespace and I deleted the next line.

**Comment by archseer** [comment]
This was discussed previously, @cessen had some thoughts about `x` in https://github.com/helix-editor/helix/issues/362

> Selecting a single line is fast and consistent and having to hold the Shift key for extending the selection is good enough (I usually extend multiple times in a row so I can just keep Shift pressed)

Yeah actually the kakoune behavior is more similar to this: `x` will always select a single line (or skip to the next one if the line is already selected) and `X` will always extend

**Comment by cessen** [comment]
> so I switched X and x which is a decent workaround

I really like this idea.  I think I'll probably switch to that myself.  Dunno if it's a better default, but for me personally I've found myself mostly using `v` + `X` when I want to select multiple lines anyway, rather than hitting `x` repeatedly.  Using `v` lets me backtrack my selection if I go too far.  So in practice I rarely take advantage of `x`'s functionality anyway.  And that would indeed fix `xd` deleting multiple lines.
